cli/azd/resources/apphost/templates/resources.bicept

```diff
@@ -337,7 +338,48 @@ resource {{bicepName $name}} 'Microsoft.DocumentDB/databaseAccounts@2023-04-15'
     }
   }
 {{end -}}
+
+{{end -}}
+
+{{range $name, $value := .SqlServers}}
+resource {{bicepName $name}} 'Microsoft.Sql/servers@2022-05-01-preview' = {
+  name: '{{$name}}-${resourceToken}'
+  location: location
+  tags: union(tags, {'aspire-resource-name': '{{$name}}'})
+  properties: {
+    minimalTlsVersion: '1.2'
+    publicNetworkAccess: 'Enabled'
+    administrators: {
+      administratorType: 'ActiveDirectory'
+      azureADOnlyAuthentication: true
+      login: managedIdentity.name
+      principalType: 'User'
+      sid: managedIdentity.properties.principalId
+      tenantId: subscription().tenantId
+    }
+  }
+}
+
+resource {{bicepName $name}}Firewall 'Microsoft.Sql/servers/firewallRules@2022-05-01-preview' = {
+    name: 'fw-{{$name}}'
+    parent: {{bicepName $name}}
+    properties: {
+      startIpAddress: '0.0.0.0'
+      endIpAddress: '255.255.255.255'
+    }
 }
+
+  {{range $cname := $value.Databases}}
+  resource {{bicepName $cname}} 'Microsoft.Sql/servers/databases@2022-05-01-preview' = {
```

Comment:
Move these 2 resources to be nested on `resource {{bicepName $name}} 'Microsoft.Sql/`, like:

```
resource {{bicepName $name}} 'Microsoft.Sql/servers@2022-05-01-preview' = {
  name: '{{$name}}-${resourceToken}'
  location: location
  tags: union(tags, {'aspire-resource-name': '{{$name}}'})
  properties: {
    minimalTlsVersion: '1.2'
    publicNetworkAccess: 'Enabled'
    administrators: {
      administratorType: 'ActiveDirectory'
      azureADOnlyAuthentication: true
      login: managedIdentity.name
      principalType: 'User'
      sid: managedIdentity.properties.principalId
      tenantId: subscription().tenantId
    }
  }
  
  {{range $cname := $value.Databases}}
  resource {{bicepName $cname}} 'servers/databases@2022-05-01-preview' = { 
   ....
  }
  
  resource {{bicepName $name}}Firewall 'servers/firewallRules@2022-05-01-preview' = {
    name: 'fw-{{$name}}'
    parent: {{bicepName $name}}
    properties: {
      startIpAddress: '0.0.0.0'
      endIpAddress: '255.255.255.255'
    }
  }
}
```

Reply:
> I was using full string because I encountered 'Error BCP029: The resource type is not valid. Specify a valid resource type of format "@".'. I'll take further exploration tomorrow on why I encountered this error.

I think you might have to change:
```
resource {{bicepName $cname}} 'servers/databases@2022-05-01-preview' = {
```

to

```
resource {{bicepName $cname}} 'databases@2022-05-01-preview' = {
```

and do a similar thing for `server/firewallRules`.  Because the resources are nested the "server" part above is implied.